src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -187,16 +189,19 @@ private async Task<bool> StopProjectAsync(HotReloadState hotReloadState, Cancell
                 {
                     await hotReloadState.Session.StopSessionAsync(cancellationToken);
 
-                    // First try to close the process nicely and if that doesn't work kill it.
-                    if (!hotReloadState.Process.CloseMainWindow())
+                    if (!hotReloadState.Process.HasExited)
                     {
-                        hotReloadState.Process.Kill();
+                        // First try to close the process nicely and if that doesn't work kill it.
+                        if (!hotReloadState.Process.CloseMainWindow())
+                        {
+                            hotReloadState.Process.Kill();
+                        }
```

Comment:
This appears to include your other PR.

Reply:
Ah, yes. Must not have switched back to `main` before creating the branch.